test/e2e/e2e.go

```diff
@@ -261,6 +261,157 @@ var _ = Describe("validation", func() {
+			}
+			// Create addresspool resource
+			By("By checking AddressPool resource is created")
+			Eventually(func() error {
+				err := testclient.Client.Get(context.Background(), key, addresspool)
+				return err
+			}, timeout, interval).Should(Succeed())
+
+			// Checking ConfigMap is created
+			By("By checking ConfigMap is created and matches the expected configuration")
+			Eventually(func() (string, error) {
+				configmap, err := testclient.Client.ConfigMaps(consts.MetallbNameSpace).Get(context.Background(), consts.MetallbConfigMapName, metav1.GetOptions{})
+				if err != nil {
+					return "", err
+				}
+				return configmap.Data[consts.MetallbConfigMapName], err
+			}, timeout, interval).Should(MatchYAML(expectedConfigMap))
+
+		},
+			table.Entry("Adding the first AddressPool object", "addresspool1", &metallbv1alpha1.AddressPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "addresspool1",
+					Namespace: consts.MetallbNameSpace,
+				},
+				Spec: metallbv1alpha1.AddressPoolSpec{
+					Name:     "test1",
+					Protocol: "layer2",
+					Addresses: []string{
+						"1.1.1.1",
+						"1.1.1.100",
+					},
+				},
+			}, `address-pools:
+- name: test1
+  protocol: layer2
+  addresses:
+
+  - 1.1.1.1
+  - 1.1.1.100
+
+`),
+			table.Entry("Adding the second AddressPool object", "addresspool2", &metallbv1alpha1.AddressPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "addresspool2",
+					Namespace: consts.MetallbNameSpace,
+				},
+				Spec: metallbv1alpha1.AddressPoolSpec{
+					Name:     "test2",
+					Protocol: "layer2",
+					Addresses: []string{
+						"2.2.2.1",
+						"2.2.2.100",
+					},
+					AutoAssign: &autoAssign,
+				},
+			}, `address-pools:
+- name: test1
+  protocol: layer2
+  addresses:
+
+  - 1.1.1.1
+  - 1.1.1.100
+
+- name: test2
+  protocol: layer2
+  auto-assign: false
+  addresses:
+
+  - 2.2.2.1
+  - 2.2.2.100
+
+`))
+
+		table.DescribeTable("Deleteing addresspools", func(addresspool *metallbv1alpha1.AddressPool, expectedConfigMap string) {
+			By("By Deleteing AddressPool CR")
+
+			Eventually(func() bool {
+				err := testclient.Client.Delete(context.Background(), addresspool)
+				return errors.IsNotFound(err)
+			}, timeout, interval).Should(BeTrue(), "Failed to delete AddressPool custom resource")
+
+			By("By checking ConfigMap matches the expected configuration")
+			Eventually(func() (string, error) {
+				configmap, err := testclient.Client.ConfigMaps(consts.MetallbNameSpace).Get(context.Background(), consts.MetallbConfigMapName, metav1.GetOptions{})
+				if err != nil {
+					// if its notfound means that was the last addresspool and configmap is deleted
+					if errors.IsNotFound(err) {
+						return "", nil
+					}
+					return "", err
+				}
+				return configmap.Data[consts.MetallbConfigMapName], err
+			}, timeout, interval).Should(MatchYAML(expectedConfigMap))
+
+		},
+			table.Entry("Deleting the first AddressPool object", &metallbv1alpha1.AddressPool{
```

Comment:
Tables are supposed to be indipendent tests that do their own beforeEach / afterEach, while here I think you are relying on them being executed in sequence.

If this is part of a single test (delete first, then delete second), there's no need for using a table.
I am not holding against this and can be changed later.